rustls/src/msgs/deframer.rs

```diff
@@ -504,9 +504,9 @@ impl DeframerBuffer<'_, InternalPayload> for DeframerVecBuffer {
 
 #[cfg(feature = "std")]
 impl<'a> DeframerBuffer<'a, ExternalPayload<'a>> for DeframerVecBuffer {
-    fn copy(&mut self, payload: &ExternalPayload<'a>, at: usize) {
+    fn copy(&mut self, payload: &ExternalPayload<'a>, _at: usize) {
         let len = payload.len();
-        self.unfilled()[at..at + len].copy_from_slice(payload.0);
+        self.unfilled()[..len].copy_from_slice(payload.0);
         self.advance(len);
     }
```

Comment:
Can someone explain this to me?

It looks like adding a `debug_assert!` would make sense here, instead of not using `at` completely. We'd compare the `at` with the `self.used` to see if they always match - and if they don't - oopsie we panic.

Reply:
> Can someone explain this to me?

Can you be more specific about what part you're not understanding? The [commit message](https://github.com/rustls/rustls/pull/1798/commits/c3335ece382f467d30150dd9e53fcd019474afba) has some additional context in the event you haven't seen it yet.

> It looks like adding a debug_assert! would make sense here, instead of not using at completely. 

Sounds reasonable. Do you want to open a PR?

Reply:
> Can you be more specific about what part you're not understanding? The [commit message](https://github.com/rustls/rustls/pull/1798/commits/c3335ece382f467d30150dd9e53fcd019474afba) has some additional context in the event you haven't seen it yet.

The motivation behind this change, the commit message explain it. It would be better capture as a code comment though.

Reply:
This looks like an very odd implementation of a trait, should it be just remodelled in a different way - in other words, should we fix the trait itself maybe?

Reply:
> in other words, should we fix the trait itself maybe?

I think we would be open to a refactoring if you see a path forward. Reviewing [CONTRIBUTING.md](https://github.com/rustls/rustls/blob/main/CONTRIBUTING.md might be helpful if so.

I understand there to be a general agreement on the team that the deframer and associated APIs have room for improvement. From my perspective it was challenging updating this area of the codebase to support the new unbuffered connection API. I've tried my hand at trying to clean up a bit and didn't arrive at anything that was a workable improvement in the time I had boxed for the task but that's likely more a product of my own relative inexperience.